python/lsst/pipe/drivers/background.py

```diff
@@ -284,7 +284,10 @@ def measureScale(self, image, skyBackground):
         statistic = afwMath.stringToStatisticsProperty(self.config.stats.statistic)
         imageSamples = []
         skySamples = []
-        for xStart, yStart, xStop, yStop in zip(xLimits[:-1], yLimits[:-1], xLimits[1:], yLimits[1:]):
+        for xIndex, yIndex in itertools.product(range(self.config.xNumSamples),
+                                                range(self.config.yNumSamples)):
+            xStart, xStop = xLimits[xIndex], xLimits[xIndex + 1]
+            yStart, yStop = yLimits[yIndex], xLimits[yIndex + 1]
```

Comment:
Do you care that all your boxes will not be the same size? Or is it just important that they are all done. Also you should put in some kind of check to make sure that self.config.(x/y)NumSamples is not greater than image.get(widht/height) or you will be sampling the same sub image over multiple times in places.

Reply:
The limits vectors (`xLimits` and `yLimits`) are set using `np.linspace`, and we're iterating over indices for those vectors. This means:
* the boxes are all (roughly) the same size (only "roughly" because the limits are integerised, so there'll be differences of up to a pixel in each dimension, but that's not a big deal)
* the iteration isn't going to run off the end of the image (the limits vectors have a length of `[xy]numSamples + 1`, we're iterating over `[xy]NumSamples` indices and retrieving the `i` and `i+1` values in each iteration, so the length and the iteration match).

Just spotted and fixed a typo.

Reply:
I was specifically worried about something like this...
```
In [7]: xNumSamples = 15

In [8]: yNumSamples = 15

In [9]: width = 10

In [10]: height = 10

In [11]: xLimits = np.linspace(0, width-1, xNumSamples+1, dtype=int)

In [12]: yLimits = np.linspace(0, height-1, yNumSamples+1, dtype=int)

In [13]: yLimits
Out[13]: array([0, 0, 1, 1, 2, 3, 3, 4, 4, 5, 6, 6, 7, 7, 8, 9])
```